controller/customdomain_utils.go

```diff
@@ -250,3 +337,68 @@ func GetInfrastructureObject(kclient client.Client) (*configv1.Infrastructure, e
 
 	return infrastructure, nil
 }
+
+// Taken from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/clusterversion.go
+// GetClusterVersionObject returns the canonical ClusterVersion object
+// To check current version: `output.Status.History[0].Version`
+//
+// `history contains a list of the most recent versions applied to the cluster.
+// This value may be empty during cluster startup, and then will be updated when a new update is being applied.
+// The newest update is first in the list and it is ordered by recency`
+//
+// Note:
+
+// Taken from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/clusterversion.go
+// SetClusterVersion sets the cluster version globally(to ENV as CLUSTER_VERSION)
+func (r *CustomDomainReconciler) GetClusterVersion(kclient client.Client) (string, error) {
+	versionObject := &configv1.ClusterVersion{}
+	ns := types.NamespacedName{
+		Namespace: "",
+		Name:      "version",
+	}
+	err := kclient.Get(context.TODO(), ns, versionObject)
+	if err != nil {
+		return "", err
+	}
+
+	// handle when there's no object defined || no version found on history
+	if len(versionObject.Status.History) == 0 || versionObject == nil {
+		return "", fmt.Errorf("version couldn't be grabbed from clusterversion: %+v", versionObject) // (%+v) adds field names
+	}
+
+	return versionObject.Status.History[0].Version, nil
+}
+
+func isUsingNewManagedIngressFeature(kclient client.Client, reqLogger logr.Logger) (bool, error) {
+	reqLogger.Info("Fetching labels from namespace", "namespace", config.OperatorNamespace)
+	ns := corev1.Namespace{}
+	if err := kclient.Get(context.TODO(), client.ObjectKey{Name: config.OperatorNamespace}, &ns); err != nil {
+		return false, err
+	}
+
+	labels := ns.Labels
+
+	return labels[legacyIngressSupportLabel] == "false", nil
+}
+
+// Taken from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/clusterversion.go
+// IsVersionHigherThan checks whether version a is higher than version b
+// input is required to be a version such as: 4.10 or 4.10.1
+// Returns false(no action) if there's an exception.
```

Comment:
```suggestion
// IsVersionGreaterOrEqualThan compares major and minor versions of a and b
// For example, if (4.10.1, 4.10.0) is supplied, this returns true because 4.10 >= 4.10
// anything other than major and minor are ignored when comparing the versions.
```

Emphasizing what the function does